### PR TITLE
Update Tag List

### LINKED ISSUE #946: [Tags deletion] When 2 tags remain and one of them is deleted, the other one is gone too
There are 4 ways to delete a tag. The described issue appears only when one of them is used

### Expected
Tag should not be deleted unless user:
- taps red 'Delete' button, or
- renames the tag to empty string, or 
- selects 'Delete' from context menu, or
- selects 'Rename' from context menu and renames tag to empty string

### Observed
Last tag is deleted if it's focused while other tag is renamed to empty string.

### Reproduced
See video below or follow the steps:
> 1. Make sure you have ideally 3 tags. Lets name them t1, t2 and t3. Can be more, but will take more time to reproduce
> 2. Swipe to the left pane and tap 'Edit' for tags
> 3. Tap t1 text. Keyboard will appear. 
> 4. Erase the text and tap t2 -> t1 will be deleted. was a bit of surprise, but understandable.
> 5. Erase text for t2 and tap t3 -> Both are deleted

IMG_0828

Make|Model|iOS Version|App Version
-|-|-|-
iPhone|XR|14.0.1|4.26

### PR DESCRIPTION
**Pull request by eshurakov**
### Details
This PR converts `SPTagsListViewController.m` to `TagListViewController.swift`
All the logic is left intact except for one moment:
* There is no longer a delay between callback from `FetchResultsController` and `tableView.reloadData()`. The delay was causing issues with data consistency as for a short (but significant) moment data in the table and data in the controller were different.
* As a result of the previous point animation of deleting a tag is gone. It will be back once we switch to ResultsController

@jleandroperez Please take a look! Thanks!

Closes #946 

Test scenarios found in previous pull requests:

### Testing: UI
1. Log into your account
2. Press over the top left button to reveal the Tags List

- [x] Verify that the `Settings` button shows up below `Trash`
- [x] Verify that Dark Mode looks as expected
- [x] Verify that Dynamic Type gets everything to properly resize
- [x] Verify that the Tags List properly remembers the previously selected cell (close and reopen)

### Testing: Tag Edition via Edition Button
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Press over the top left button to reveal the Tags List
5. Press the `Edit` button

- [x] Verify that **only** tag rows can be moved
- [x] Verify that **only** tag rows can be deleted
- [x] Verify that **only** tag rows can be renamed
- [x] Verify that while a Tag row is being renamed, the tableView's bottom insets are properly adjusted: you should be able to scroll down to the bottom.

### Testing: Tag Edition via MenuController
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Perform a long press gesture over any of the tags

- [x] Verify that the highlighted tag can be renamed
- [x] Verify that the highlighted tag can be deleted
- [x] Verify that a long press over `All Notes` / `Trash` / `Settings` do not trigger such menu

### Testing: Renaming
1. Make sure you have 3 or more tags
2. Open Sidebar
3. Rename first tag to an empty string
4. Instead of pressing `Done`, press on the second tag

- [x] Verify that the first tag is gone
- [x] Verify that the second tag is still in the list

### Testing: Tags Sort Order
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Press over the top left button to reveal the Tags List
5. Press over `Settings`
6. Change the `Tags` sort order

- [x] Verify that the Tags List gets properly updated

### Testing: Hiding Tags Sections
1. Log into a testing account with no Tag

- [x] Verify that an account with no Tags doesn't show the `Tags` section
- [x] Verify that an account with no Tags doesn't show the `Untagged` section

### Testing: Predicates!
1. Log into your account
2. Add three notes (with random contents)
3. Tag one of the new notes
4. Trash one of the new notes
5. And of course, leave one of them untagged

- [x] Verify the `Trash` filter shows the trashed note
- [x] Verify the `Untagged` filter shows the untagged note
- [x] Verify that tapping over the tag you've set in (3) properly displays the new note

### CODE REVIEW

File: Simplenote/TagListViewController.swift [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+            cell.textField.text = NSLocalizedString("Settings", comment: "")
+            cell.iconImage = .image(name: .settings)
+            cell.accessibilityIdentifier = "settings"
+        }
+    }
+
+    func configureTagCell(_ cell: SPTagListViewCell, at indexPath: IndexPath) {
+        let tagName = tag(at: indexPath)?.name
+
+        cell.textField.text = tagName
```
Comment (jleandroperez) [inline Simplenote/TagListViewController.swift:390]:
```suggestion
        cell.textField.text = tagName
        cell.textField.delegate = self
```

Comment (jleandroperez) [reply Simplenote/TagListViewController.swift:390]: Missing textField.delegate initialization :fire:

File: Simplenote/SPAppDelegate.h [hunk 2]
```diff
@@ -22,7 +22,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (strong, nonatomic, readonly) NSPersistentStoreCoordinator	*persistentStoreCoordinator;
 
 @property (strong, nonatomic) SPSidebarContainerViewController          *sidebarViewController;
-@property (strong, nonatomic) SPTagsListViewController                  *tagListViewController;
+@property (strong, nonatomic) TagListViewController                  *tagListViewController;
```
Comment (jleandroperez) [inline Simplenote/SPAppDelegate.h]: (SORRY!!!) Nipticky: Indentation 😂 

(Seriously, apologies!!)

File: Simplenote/TagListViewController.swift (hunk 1 above)
Comment (jleandroperez) [reply Simplenote/TagListViewController.swift:390]: ^^^ Please disregard, I've noticed you're explicitly setting / resetting the delegate when needed!

File: Simplenote/TagListViewController.swift [hunk 3]
```diff
@@ -0,0 +1,746 @@
...
+
+        guard let indexPath = tableView.indexPath(for: cell) else {
+            return
+        }
+
+        removeTag(at: indexPath)
+    }
+}
+
+// MARK: -
```
Comment (jleandroperez) [inline Simplenote/TagListViewController.swift]: Nipticky: Missing comment

**Review by jleandroperez — APPROVED** [review]
Looks solid!!! Thank you for doing this @eshurakov !!

Please feel free to :shipit: once the merge conflict is resolved.

🔥 🔥 🔥

File: Simplenote/SPAppDelegate.h (hunk 2 above)
Comment (eshurakov) [reply Simplenote/SPAppDelegate.h]: Fixed 👍

**Comment by eshurakov** [comment]
Thank you for the review!

**Pull request merged**